server/src/main/java/org/elasticsearch/client/node/NodeClient.java

```diff
@@ -79,7 +79,13 @@ public void close() {
     public <Request extends ActionRequest, Response extends ActionResponse>
     void doExecute(ActionType<Response> action, Request request, ActionListener<Response> listener) {
         // Discard the task because the Client interface doesn't use it.
-        executeLocally(action, request, listener);
+        try {
+            executeLocally(action, request, listener);
+        } catch (Exception e) {
```

Comment:
I think without further refactoring of the signature of `registerAndExecute` and this is about as low as we can go on the stack in terms of passing these exceptions to the listener directly. Otherwise we'll have to adjust a non-trivial number of spots that make use of the returned `Task` from that method.

Reply:
I think we can't just catch that exception here since we also have the case of `org.elasticsearch.tasks.TaskManager#register` throwing `IllegalArgumentException` at least (that actually does reproduce in a test even). Also, technically speaking the in-line task cancellation in `org.elasticsearch.tasks.TaskManager#registerCancellableTask` could throw any exception if the parent is banned already so I figured I'd just catch all the things here for now?